### DISCUSSION TITLE
popup opens not in every case

### DISCUSSION THREAD

**Discussion by manuel-koch** [AllenDang/giu#623] [Q&A] [answered]
Trying to open a popup, but it only works in one of three code paths.

What am I doing wrong ?

```go
// PopupName is a global constant string variable

giu.SingleWindowWithMenuBar().Layout(
  
  giu.PopupModal(PopupName).Layout(
    giu.Label("Hello"),
    giu.Button("Ok").OnClick(func() { giu.CloseCurrentPopup() }),
  ),
  
  giu.MenuBar().Layout(
    giu.Menu("Help").Layout(
      giu.MenuItem("test a").OnClick(func() {
        // DOESN'T WORK
        giu.OpenPopup(PopupName)
      }),
    ),
  ),
  
  giu.Selectable("test b").OnClick(func() {
    // DOES WORK
    giu.OpenPopup(PopupName)
  }),
  
  giu.Label("foo"),
  giu.ContextMenu().Layout(
    giu.Selectable("test c").OnClick(func() {
      // DOESN'T WORK
      giu.OpenPopup(PopupName)
    }),

  ....
),
```

**Comment by AllenDang** [comment, marked as answer]
Popup can only be opened by the widget which be defined at the same level.
In your code, menuitem is inside menu, and selectable is inside context menu.
To ignore this mechanism, you could use a boolean to open the popup instead.

**Reply by manuel-koch** [reply]
but doesn't this ( using a boolean instead of a string ) imply that I can only have one popup in the whole app/window ?

**Reply by AllenDang** [reply]
Yes, you can, thats how msgbox works

**Reply by gucio321** [reply]
@AllenDang @manuel-koch in my projects I'm using a simple wrapper that makes popup management easier
An insurance if this file is e.g. in https://github.com/thegreaterheptavirate/constitutor in pkg/app package

I didn't want to pr it to giu since it doesn't meet its convention of "factories"
